### PR TITLE
CORE-7075 - rename health monitor

### PR DESCRIPTION
**Pull request by driessamyn**
Renamed Health Monitor to Worker Monitor as it also serves metrics.

### CODE REVIEW

File: applications/workers/worker-common/src/main/kotlin/net/corda/applications/workers/workercommon/WorkerMonitor.kt [hunk 1]
```diff
@@ -1,15 +1,16 @@
...
  *
  * A worker indicates its healthiness/readiness by returning a 200 code for HTTP requests to
  * `HTTP_HEALTH_ROUTE`/`HTTP_READINESS_ROUTE`.
+ * Worker metrics are reported on `HTTP_STATUS_ROUTE`.
  *
  * A worker is considered healthy if no component has a `LifecycleStatus` of `LifecycleStatus.ERROR`. A worker is
  * considered ready if no component has a `LifecycleStatus` of either `LifecycleStatus.DOWN` or `LifecycleStatus.ERROR`.
  */
-interface HealthMonitor {
+interface WorkerMonitor {
```
Comment (kyriathar) [inline applications/workers/worker-common/src/main/kotlin/net/corda/applications/workers/workercommon/WorkerMonitor.kt:13]: Do we also need to keep `MetricsServer` interface?

File: applications/workers/worker-common/src/main/kotlin/net/corda/applications/workers/workercommon/WorkerMonitor.kt [hunk 2]
```diff
@@ -1,15 +1,16 @@
 package net.corda.applications.workers.workercommon
 
 /**
- * Monitors the health of a worker.
+ * Exposes an HTTP endpoint to report health, status and metrics for the worker.
  *
  * A worker indicates its healthiness/readiness by returning a 200 code for HTTP requests to
  * `HTTP_HEALTH_ROUTE`/`HTTP_READINESS_ROUTE`.
+ * Worker metrics are reported on `HTTP_STATUS_ROUTE`.
```
Comment (kyriathar) [inline applications/workers/worker-common/src/main/kotlin/net/corda/applications/workers/workercommon/WorkerMonitor.kt]: Should it be `HTTP_METRICS_ROUTE`?

**Review by kyriathar — COMMENTED** [review]
Some minor comments

File: applications/workers/worker-common/src/main/kotlin/net/corda/applications/workers/workercommon/WorkerMonitor.kt (hunk 1 above)
Comment (driessamyn) [reply applications/workers/worker-common/src/main/kotlin/net/corda/applications/workers/workercommon/WorkerMonitor.kt:13]: Good spot. That shouldn't ever have been in there :(

File: applications/workers/worker-common/src/main/kotlin/net/corda/applications/workers/workercommon/WorkerMonitor.kt (hunk 2 above)
Comment (driessamyn) [reply applications/workers/worker-common/src/main/kotlin/net/corda/applications/workers/workercommon/WorkerMonitor.kt]: Also good spot. Thanks.

**Review by vkolomeyko — APPROVED** [review]
LGTM

**Review by kyriathar — APPROVED** [review]

**Review by davidcurrie — APPROVED** [review]
🚀

**Pull request merged**